### DISCUSSION TITLE
MacOS

### DISCUSSION THREAD

**Discussion by cigas4** [thirtythreeforty/neolink#135] [Q&A]
Hellow, how can I install MacOS program? When I download it, I get folder with two files in it, one is without any extension, other is ".d". None of them is doing anything... 
Also is it possible to run Neolink together with Reolink software? From description I see it is recomended to use Blue Iris software which is paid, can I use any other free-ware to run Neolink (if Reolink software is not supported)?

**Comment by QuantumEntangledAndy** [comment]
The one without the extension is the program. It's a terminal application and you need to be familiar with the terminal app to use it. 

I don't use blue-iris. I use other rtsp programs like ffmpeg and VLC. It really dosent matter. Rtsp is a standard format and supported by lots of programs.

Your comment about reolink app is odd. It makes me wonder if your aware of what neolink is.

Neolink is a translator, it takes as input data from the camera and outputs data as an rtsp stream. It is then up to you to decide how to open and use the stream.

Perhaps you could tell me your intended use of the program and I can best direct how to help you.

**Reply by cigas4** [reply]
Hi,

Thanks for your replay. Yeah, I am not really familiar with this kind of installation through the terminal so your help would be really appreciated.
Regarding the usage, I bought this camera (Reolink Lumus) to work with Home Assistant. However I didn’t checked that it does not support RTSP stream when buying it.. So from HA group I got suggestion to use this program to be able to get RTSP and be able to integrate it to HA.
Now it gets a bit clear, because at first I thought I need to run Reolink software on my computer and then Neolink is converting this stream to RTSP. 
Another option for me is Linux Debian. Now I am trying to get another instance of Home Assistance running on VirtualBox (on macos). I will have virtual machine running all the time anyway, so it depends if there is any difference (in terms of installation difficulty) to have Neolink running on MacOS directly or on Debian VM.

**Reply by QuantumEntangledAndy** [reply]
I think Debian is a little easier to setup. But that's because I like the terminal :). Anyways it's up to you.

If you have a raspberry pi I would recommend you install home assistant and neolink together on that so that it's always running.


---

# macos

For Mac go the gstreamer website and download and install that. 

Download and extract the neolink zip to the downloads folder.

From here.... I need to know if you have the folder `/usr/local/bin` or not. If you don't have it you should make it. If you do skip this step.

Copy the neolink program from the zip file to `/usr/local/bin/neolink`.

Open up the terminal app. It's in `/Applications/Utilities/Terminal.app`

Run this command

```bash
chmod +x /usr/local/bin/neolink
```

Now run

```
neolink --help
```

If it all worked you will get the neolink help message. If not show me pictures of what failed.


---

# Debian

Open a terminal and run this

```bash
sudo apt install \
    libgstrtspserver-1.0-0 \
    libgstreamer1.0-0 \
    libgstreamer-plugins-bad1.0-0 \
    gstreamer1.0-plugins-good \
    gstreamer1.0-plugins-bad
```

Download and extract neolink zip

In the terminal run

```bash
sudo cp ~/Downloads/neolink /usr/local/bin/neolink
sudo chmod +x /usr/local/bin/neolink
neolink --help
```

If all worked you will get the neolink help message if not send me pictures.

**Reply by cigas4** [reply]
Hello,

Finally up and running.. I have installed it on Linux VM at the moment to play around, because my main HA is OS version, so I can't run enything else on my rpi4. Also found your instructions for making service file to have auto-start, also works good. Thanks alot!

**Reply by QuantumEntangledAndy** [reply]
I see well drop us another line if you need more questions answered. 

I think you should be able to use the SSH server add-on to get a shell on the has os and then you can install it on from there. Might want to consider the non has os option I did that for my one and it's really quite easy.

**Reply by cigas4** [reply]
Hi,

I just did new setup of linux installation on my rpi4 and unfortunately cant get neolink working anymore.
Processor: BCM2711
Distribution: Raspbian GNU/Linux 10 (buster)
Kernel version: Linux 5.10.17-v7l+ armv7l
Firmware: #1403

I have downloaded release-i386-buster archive and during the steps I get this message:

```
pi@RPI-NVR:~ $ sudo apt install \
>     libgstrtspserver-1.0-0 \
>     libgstreamer1.0-0 \
>     libgstreamer-plugins-bad1.0-0 \
>     gstreamer1.0-plugins-good \
>     gstreamer1.0-plugins-bad
Reading package lists... Done
Building dependency tree       
Reading state information... Done
gstreamer1.0-plugins-bad is already the newest version (1.14.4-1+rvt+deb10u2).
libgstreamer-plugins-bad1.0-0 is already the newest version (1.14.4-1+rvt+deb10u2).
libgstreamer1.0-0 is already the newest version (1.14.4-1).
libgstrtspserver-1.0-0 is already the newest version (1.14.4-1).
gstreamer1.0-plugins-good is already the newest version (1.14.4-1+rpt1).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
pi@RPI-NVR:~ $ sudo cp ~/Downloads/neolink /usr/local/bin/neolink
pi@RPI-NVR:~ $ sudo chmod +x /usr/local/bin/neolink
pi@RPI-NVR:~ $ neolink --help
-bash: /usr/local/bin/neolink: cannot execute binary file: Exec format error
pi@RPI-NVR:~ $
```

Do you have any ideas?

**Reply by QuantumEntangledAndy** [reply]
yeah raspberry pi is armv7 not i386. You want the armhf version [here](https://nightly.link/thirtythreeforty/neolink/workflows/build/master/release-armhf-buster.zip)

**Reply by cigas4** [reply]
Hi,

Thanks for your blazing fast answer. That was exactly my problem... Still it is strange since I have saved all the files to achive from last installation together with all the instuctions.. Anyway, thanks.
